### ISSUE TITLE
Search tags allow list

### ISSUE THREAD

**Issue by lmickh** [grafana/tempo#1362]
**Is your feature request related to a problem? Please describe.**
Finding large or troublesome tags to put into `SearchTagsDenyList` is not easy.  Users might have a list of tags they know covers the bulk of their search needs that doesn't need to include dynamic data that can often find its way into tags. It would be nice to set an allow list to limit the scope of the search data as opposed to just a deny list.

**Describe the solution you'd like**
Provide an option called `search_tags_allow_list` that functions similar to `search_tags_deny_list`.  Only tags that matched the list would generate the related search data.  An empty list would disable the allow list filtering.

This option would either have to conflict with or override the deny list.  If both were given, it would make sense to warn that the deny list is superseded by the allow list.  

**Describe alternatives you've considered**
Possibly provide an option to limit searches to just the parent trace as a way to reduce the overhead, but that can cause its own set of problems as well.

**Additional context**

**Comment by joe-elliott** [comment]
We are likely moving to a new block type that will change the way search works quite a bit. So it may not make sense to invest in this.

Long term this may still apply to recent traces though. @mdisibio what do you think?

**Comment by lmickh** [comment]
I played around with adding this, but I think we can hold off on a PR until the research on the new block type is complete.  I don't mind spending time on it, but I don't want to introduce a feature/option that has to be carried forward when the new block type might make it a non-issue.

**Comment by joe-elliott** [comment]
To be clear the newer block format is probably 6 months or so out. If this is an important feature to you in the short term it should be a relatively easy addition. I would be glad to merge it.

**Comment by lmickh** [comment]
Its not an immediate need at the moment now that I've got the deny list built up already..  I figured out that `db.statement` was a big culprit to the tag size.

I haven't had a lot of time to spend on this, but what I've done so far is [on this branch](https://github.com/lmickh/tempo/tree/search-tags-allow-list).  I still need to sort out getting testing setup.  If someone wants to run with it, you won't hurt my feelings.
